### PR TITLE
Search result groups

### PR DESCRIPTION
**Pull request by chigby**
Groups search results into groups by content type on the search result page

Refs #1562

### CODE REVIEW

**Review by nmorduch — CHANGES_REQUESTED** [review]
I'm running into a problem where, because this categorizes whatever comes in on a given page of search results, the first page has a lot of programs and no people, and the second page has a lot of people. I probably described unclearly what I was looking for, which was that it should have these 5 sections, and then under each section we should return the first x results if that section has any results at all (which was why I was imagining doing it on the backend, since that's where the pagination is). I'll update the issue with a mockup that gives a sense of how many of each type I'm looking to see (since i think it will be different for different types) along with any ordering we want within the category (which, to be fair, is just now occurring to me as being desirable).

File: newamericadotorg/assets/react/home-panels/pages/Search.js [hunk 1]
```diff
@@ -6,6 +6,37 @@ import * as QueryString from 'query-string';
...
+      collection.push(item);
+    }
+  });
+  return map;
+}
+
+function classifyResult(result) {
+  let type = result.content_type.api_name;
+
+  if (type == "article" || type == "blogpost" || type == "book" || type == "event" || type == "issueortopic" || type == "podcast" || type == "policypaper" || type == "pressrelase" || type == "quoted" || type == "weeklyarticle" || type == "report" || type == "indepth") {
```
Comment (nmorduch) [inline newamericadotorg/assets/react/home-panels/pages/Search.js:27]: Looks like events are going into this category so not making it to the "events" category, and "issueortopic" here should go into "other pages"

**Comment by chigby** [comment]
@nmorduch Updated this PR to be more in line with what you're describing. The search page now separately searches each of the buckets and returns separately-paginated result buckets.

**Comment by chigby** [comment]
Further updated the PR to reduce the number of results per page on each bucket to 6. Also made it customizable per-bucket.

**Comment by nmorduch** [comment]
@chigby can you review my commit? and this should be deployed to the dev server soon

**Review by nmorduch — APPROVED** [review]

**Pull request merged**